File: apps/cli/cmd/system.go
package cmd

import (
	"github.com/spf13/cobra"
	"yishan/apps/cli/internal/output"
	cliruntime "yishan/apps/cli/internal/runtime"
)

var healthCmd = &cobra.Command{
	Use:   "health",
	Short: "Check API health",
	Long:  `Check the health of the Yishan API service. Exits with a non-zero code if the API is unreachable.`,
	Example: `  yishan health
  yishan health --output json`,
	RunE: func(_ *cobra.Command, _ []string) error {
		response, err := cliruntime.APIClient().Health()
		if err != nil {
			return err
		}

		return output.PrintAny(response)
	},
}

var whoamiCmd = &cobra.Command{
	Use:     "whoami",
	Aliases: []string{"me"},
	Short:   "Show current authenticated user",
	Long:    `Show the identity of the currently authenticated user based on the active API token.`,
	Example: `  yishan whoami
  yishan whoami --output json`,
	RunE: func(_ *cobra.Command, _ []string) error {
		response, err := cliruntime.APIClient().WhoAmI()
		if err != nil {
			return err
		}

		return output.PrintAny(response)
	},
}

func init() {
	rootCmd.AddCommand(healthCmd)
	rootCmd.AddCommand(whoamiCmd)
}

File: apps/cli-rs/src/cmd/system.rs
use crate::output::print_any;
use crate::runtime::AppRuntime;
use clap::Subcommand;

#[derive(Subcommand)]
pub enum SystemCommands {
    /// Check API health
    Health,
    /// Show authenticated user information
    Whoami,
}

pub async fn run(cmd: SystemCommands, runtime: &AppRuntime) -> anyhow::Result<()> {
    match cmd {
        SystemCommands::Health => {
            let client = runtime.api_client();
            let resp = client.health().await?;
            print_any(resp)
        }
        SystemCommands::Whoami => {
            let client = runtime.api_client();
            let resp = client.whoami().await?;
            print_any(resp)
        }
    }
}
